File: c/10.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

int main()
{
    static long target = 2000000;
    long long total = 0;
    int primes[target];
    long i, j;
    for (i=0; i<= target; i++) {
        primes[i] = true;
    }

    for (i=2; i < sqrt(target); i++) {
        if (primes[i] == true) {
            for (j=i*2; j<target; j+=i) {
                primes[j] = false;
            }
        }
    }

    for (i=2; i < target; i++) {
        if (primes[i] == true) {
            total += i;
        }
    }

    printf("%lli\n", total);
}

File: rust/src/bin/10.rs
fn main() {
    let target = 2000000;
    let mut total = 0;
    let mut primes = vec![true; target];

    let max_target = (target as f64).sqrt();
    for i in 2..(max_target as usize) {
        if primes[i] {
            let mut j = i*2;
            while j < target {
                primes[j] = false;
                j += i;
            }
        }
    }
    for (i, prime) in primes.iter().enumerate().take(target).skip(2) {
        if *prime {
            total += i;
        }
    }
    println!("{}", total);
}
